allennlp/predictors/multitask.py

```diff
@@ -0,0 +1,102 @@
+from overrides import overrides
+from typing import Type, List, Dict
+
+from allennlp.common import JsonDict
+from allennlp.data import DatasetReader, Instance
+from allennlp.models import Model
+from allennlp.predictors.predictor import Predictor
+
+
+@Predictor.register("multitask")
+class MultiTaskPredictor(Predictor):
+    """
+    Predictor for multitask models.
+
+    Registered as a `Predictor` with name "multitask".
+    """
+
+    _WRONG_READER_ERROR = (
+        "MultitaskPredictor is designed to work with MultiTaskDatasetReader. "
+        + "If you have a different DatasetReader, you have to write your own "
+        + "Predictor, but you can use MultitaskPredictor as a starting point."
+    )
+
+    _WRONG_FIELD_ERROR = (
+        "MultiTaskPredictor expects instances that have a MetadataField "
+        + "with the name 'task', containing the name of the task the instance is for."
+    )
+
+    def __init__(self, model: Model, dataset_reader: DatasetReader) -> None:
```

Comment:
Can we specify the type restriction in the type hints too?

```suggestion
    def __init__(self, model: MultiTaskModel, dataset_reader: MultiTaskDatasetReader) -> None:
```

Reply:
Turns out we can! Yay!